### DISCUSSION TITLE
Only show whisper responses in the split the whisper was executed in.

### DISCUSSION THREAD

**Discussion by mmattbtw** [Chatterino/chatterino2#3649] [Ideas]
Right now, whispers show up in all tabs/splits. This means that whispers can flood a chat that might not be as active.

<img width="527" alt="image" src="https://user-images.githubusercontent.com/30363562/161445446-eb3cb3f7-4253-455f-8919-21d1e25c170d.png">
This is the current behavior, the whisper was executed in the left split, however also showed up on the right. The behavior I am thinking of is that the whisper only shows in the executed tab (left split in this case).

**Comment by pajlada** [comment]
We cannot know what outgoing whisper triggered an incoming whisper.
The way you could handle it instead is:
1) Only show incoming whispers in the currently active split
2) Only show incoming whispers in the split you last sent a whisper from. This could be troublesome since we wouldn't know where to start showing whispers

Another solution that already exists right now in Chatterino is to use the `/whispers` split and disable the "Show Twitch whispers inline" option
![image](https://user-images.githubusercontent.com/962989/162573602-da8272ed-412a-45e6-9e77-bd130b67e0d1.png)
Then it would show up like this: 
![image](https://user-images.githubusercontent.com/962989/162573617-40ffeb79-01f9-4366-8024-91ce3a83e66c.png)

**Reply by LosFarmosCTL** [reply]
I could see a 3rd option where you try to be clever about the conditions of when to treat an incoming message as a response and where to show it: i.e. if you send a message to someone in one chat, and that exact person responds to you it would only show up in the chat you sent the first whisper from. This would also introduce the problems from option 2 though, with the additional question of when you would count an incoming whisper as a response (lets say you send a whisper and that same person either sends you something unrelated 1-2 hours later, or just takes a long time to respond, sending that message to a chat that you were active in 1-2 hours earlier would 100% result in confusion for a lot of users)

That said, personally I think the first option might be the most "unspecial", but could solve the issue of whispers cluttering up your entire chatterino, while avoiding the possible confusion other options could result in.

**Reply by iProdigy** [reply]
option 1 would be a nice setting to have (but default disabled due to bad UX if users are frequently switching splits and whispering to the same target)

**Comment by brian6932** [comment]
I like this idea, and wanted to add another possible approach. A notification toast sent on receiving a `whisper`, which on click, takes you to the `/whispers` split, and if you already have `/whispers` (and the window itself) focused, the notification toasts should be suppressed.
